packages/logger/tests/unit/logEvent.test.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Logger log event tests
+ *
+ * @group unit/logger/logger/logEvent
+ */
+import context from '@aws-lambda-powertools/testing-utils/context';
+import { beforeEach, describe, expect, it, jest } from '@jest/globals';
+import middy from '@middy/core';
+import type { Context } from 'aws-lambda';
+import { Logger } from '../../src/Logger.js';
+import { injectLambdaContext } from '../../src/middleware/middy.js';
+
+const event = {
+  foo: 'bar',
+};
+
+const logSpy = jest.spyOn(console, 'info');
+
+describe('Log event', () => {
+  const ENVIRONMENT_VARIABLES = process.env;
+
+  beforeEach(() => {
+    process.env = {
+      ...ENVIRONMENT_VARIABLES,
+      POWERTOOLS_LOGGER_LOG_EVENT: 'true',
+      POWERTOOLS_DEV: 'true',
+    };
+    jest.resetAllMocks();
+  });
+
+  it('logs the event with the correct log level and message', () => {
+    // Prepare
+    const logger = new Logger();
+
+    // Act
+    logger.logEventIfEnabled(event);
+
+    // Assess
+    expect(logSpy).toHaveBeenCalledTimes(1);
+    expect(JSON.parse(logSpy.mock.calls[0][0])).toStrictEqual(
+      expect.objectContaining({ event })
+    );
+  });
+
+  it("doesn't log the event when the feature is disabled", () => {
+    // Prepare
+    process.env.POWERTOOLS_LOGGER_LOG_EVENT = 'false';
+    const logger = new Logger();
+
+    // Act
+    logger.logEventIfEnabled(event);
+
+    // Assess
+    expect(logSpy).not.toHaveBeenCalled();
+  });
+
+  it('respects the overwrite flag when provided', () => {
+    // Prepare
+    const logger = new Logger();
+
+    // Act
+    logger.logEventIfEnabled(event, false);
+
+    // Assess
+    expect(logSpy).not.toHaveBeenCalled();
+  });
+
+  it('logs the event when logEvent is set in the Middy.js middleware', async () => {
+    // Prepare
+    const logger = new Logger();
+    const handler = middy(async () => {}).use(
+      injectLambdaContext(logger, { logEvent: true })
+    );
+
+    // Act
+    await handler(event, context);
+
+    // Assess
+    expect(logSpy).toHaveBeenCalledTimes(1);
```

Comment:
Need this?

Reply:
See answer [here](https://github.com/aws-powertools/powertools-lambda-typescript/pull/2942/files/55a2b1e6fee4469c07d280c5d3e9d2df83f836b8#r1722048526).